mm2src/mm2_main/src/rpc/lp_commands/lr_swap/lr_impl.rs

```diff
@@ -0,0 +1,357 @@
+                lr_data
+                    .dst_contract
+                    .as_ref()
+                    .ok_or(ApiIntegrationRpcError::InternalError("no contract".to_owned()))?,
+            );
+            let chain_id = lr_data
+                .chain_id
+                .ok_or(ApiIntegrationRpcError::InternalError("no chain id".to_owned()))?;
+            let query_params = ClassicSwapQuoteParams::new(src_contract, dst_contract, src_amount.to_string())
+                .with_include_tokens_info(Some(true))
+                .with_include_gas(Some(true))
+                .build_query_params()
+                .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, lr_data.dst_decimals))?;
+            let api_client = ApiClient::new(ctx)
+                .mm_err(|api_err| ApiIntegrationRpcError::from_api_error(api_err, lr_data.dst_decimals))?;
+            let fut = api_client.call_one_inch_api::<ClassicSwapData>(
+                Some(chain_id),
+                ApiClient::classic_swap_endpoint(),
+                ApiClient::quote_method().to_owned(),
+                Some(query_params),
+            );
+            quote_futs.push(fut);
+            src_dst.push((src_token.clone(), dst_token.clone()));
+        }
+
+        let swap_data = join_all(quote_futs)
+            .await
+            .into_iter()
+            .collect::<MmResult<Vec<_>, _>>()
+            .mm_err(|cli_err| ApiIntegrationRpcError::from_api_error(cli_err, None))?;
+        let swap_data_map = src_dst
+            .into_iter()
+            .zip(swap_data.into_iter())
+            .collect::<HashMap<(_, _), _>>();
+        self.update_with_lr_swap_data(&swap_data_map);
+        Ok(())
+    }
+
+    /// Select the best swap path, by minimum of total swap price (including order and LR swap)
+    #[allow(clippy::result_large_err)]
+    fn select_best_swap(&self) -> MmResult<(ClassicSwapData, RpcOrderbookEntryV2, MmNumber), ApiIntegrationRpcError> {
+        // Calculate swap's total_price (filling the order plus LR swap) as src_amount / order_amount
+        // where src_amount is user tokens to pay for the swap with LR, 'order_amount' is amount which will fill the order
+        // Tx fee is not accounted here because it is in the platform coin, not token, so we can't compare LR swap tx fee directly here.
+        // Instead, GUI may calculate and show to the user the total spendings for LR swap, including fees, in USD or other fiat currency
+        let calc_total_price = |src_amount: U256, lr_swap: &ClassicSwapData, order: &RpcOrderbookEntryV2| {
+            let src_amount = mm_number_from_u256(src_amount);
+            let order_price = MmNumber::from(order.price.rational.clone());
+            let dst_amount = MmNumber::from(lr_swap.dst_amount.to_string().as_str());
+            if let Some(order_amount) = dst_amount.checked_div(&order_price) {
+                let total_price = src_amount.checked_div(&order_amount);
```

Comment:
nit: that looked not very intuitive IMO. we better get `lr_price = dst/src` and `order_price` provided by the order and multiply them together.